### DISCUSSION TITLE
State of GRPC with rust-protobuf?

### DISCUSSION THREAD

**Discussion by mwlon** [stepancheg/rust-protobuf#613] [Q&A] [answered]
@stepancheg I really appreciate your work, and I'm curious about the present and future of GRPC with rust-protobuf:

- In the project's current state, are there any good GRPC clients that work with it? I know you've previously worked on [a GRPC implementation](https://crates.io/crates/grpc), but it sounds like it's not ready for production use.
- Are there any examples of using rust-protobuf with a GRPC client?
- I know you've been looking for more community support in developing this project. Does it look like you or others will be able to get your grpc crate to a completed state anytime soon?

Thanks.

**Comment by stepancheg** [comment, marked as answer]
> are there any good GRPC clients that work with it? 

grpc-rs, which depends on native C (or C++?) implementation

And there's tonic-grpc for prost protobuf. It should be possible to make tonic-grpc work with rust-protobuf, but nobody did it yet.

> on [a GRPC implementation](https://crates.io/crates/grpc), but it sounds like it's not ready for production use

Right.

> Are there any examples of using rust-protobuf with a GRPC client?

grpc-rs or grpc-rust should have examples.

> Does it look like you or others will be able to get your grpc crate to a completed state anytime soon?

Soon, definitely not.

**Comment by 0xcaff** [comment]
I've gotten GRPC working with tonic (codegen and all). I've added it as an example to tonic. https://github.com/hyperium/tonic/pull/1789 The setup is a little big kludgy but the usage works great!
